src/Http/GraphRequestUtil.php

```diff
@@ -30,16 +30,27 @@ class GraphRequestUtil
      * @param string $baseUrl if empty, is overwritten with $client's national cloud
      * @param string $endpoint can be a full URL
      * @param string $apiVersion
-     * @return UriInterface|null
+     * @return UriInterface
+     * @throws \InvalidArgumentException
      */
-    public static function getRequestUri(string $baseUrl, string $endpoint, string $apiVersion = "v1.0"): ?UriInterface {
-        // If endpoint is a full url, ensure the host is a national cloud or custom host
+    public static function getRequestUri(string $baseUrl, string $endpoint, string $apiVersion = "v1.0"): UriInterface {
+        // If endpoint is a full url, ensure the host is a national cloud
         if (parse_url($endpoint, PHP_URL_SCHEME)) {
-            return (NationalCloud::containsNationalCloudHost($endpoint)) ? new Uri($endpoint) : null;
+            if (NationalCloud::containsNationalCloudHost($endpoint)) {
+                return new Uri($endpoint);
+            }
+            throw new \InvalidArgumentException("Invalid national cloud host in endpoint=".$endpoint.". See https://docs.microsoft.com/en-us/graph/deployments#microsoft-graph-and-graph-explorer-service-root-endpoints.");
         }
-        $relativeUrl = (NationalCloud::containsNationalCloudHost($baseUrl)) ? "/".$apiVersion : "";
-        $relativeUrl .= (substr($endpoint, 0, 1) == "/") ? $endpoint : "/".$endpoint;
-        return UriResolver::resolve(new Uri($baseUrl), new Uri($relativeUrl));
+        if ($baseUrl) {
+            $urlParts = parse_url($baseUrl);
+            if (!$urlParts || !array_key_exists("scheme", $urlParts) || !array_key_exists("host", $urlParts)) {
```

Comment:
There is a similar check in `NationalCloud::containsNationalCloudHost($baseUrl))` so we could use it to validate here, I think.

Reply:
I had added this validation separately to cover the case where customers pass custom base URL's when setting up the request.
I can make the National cloud host validation & custom base Url validation cleaner though since there are similar checks across both.

Reply:
- Refactored this to prevent parsing the url twice.
- It wouldn't be possible to only use `NationalCloud::containsNationalCloudHost()` to validate a base url since we need to support non-graph base urls (https://github.com/microsoftgraph/msgraph-sdk-design/pull/45)